core/src/main/java/org/apache/iceberg/BaseRewriteFiles.java

```diff
@@ -57,4 +64,11 @@ public RewriteFiles rewriteFiles(Set<DataFile> filesToDelete, Set<DataFile> file
 
     return this;
   }
+
+  @Override
+  public RewriteFiles rewriteSpec(PartitionSpec newSpec) {
+    Preconditions.checkArgument(current().specsById().containsKey(newSpec.specId()),
+        "Invalid spec with id %d", newSpec.specId());
+    return new BaseRewriteFiles(tableName(), ops(), newSpec);
```

Comment:
Not returning `this` breaks method chaining because you can't do this:

```java
RewriteFiles rewrite = table.newRewriteFiles();

if (!spec.equals(table.spec())) {
  rewrite.rewriteSpec(spec);
}

rewrite.commit();
```